android-studio-for-platform: init at 2023.2.1.20

robbins opened the pull request:
## Description of changes

`android-studio-for-platform` is a variant of the Android Studio IDE designed specifically for working on AOSP.
homepage: https://developer.android.com/studio/platform

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

robbins committed:
android-studio-for-platform: init at 2023.2.1.20

robbins commented:
Had a previous PR open here: https://github.com/NixOS/nixpkgs/pull/269108 but forgot about it and deleted my fork of `nixpkgs`, which closed the pull request.

I will be re-testing to see if the errors I mentioned in the comments of that PR still exist or if they impact any functionality.

BenediktBroich commented:
Result of `nixpkgs-review pr 278361` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>android-studio-for-platform</li>
  </ul>
</details>

robbins commented:
> Had a previous PR open here: #269108 but forgot about it and deleted my fork of `nixpkgs`, which closed the pull request.
> I will be re-testing to see if the errors I mentioned in the comments of that PR still exist or if they impact any functionality.

Fixed the Soong sync errors found there.

shymega commented:
If you need testers for this PR, let me know. I'm running NixOS on x86_64 Linux, and have several AOSP trees to work with. I have around 32GB of RAM.

I was just about to start work on a branch, but no need now. Good luck!

robbins committed:
androidStudioForPlatformPackages.canary: init at 2023.3.2.1

Artturin commented on pkgs/applications/editors/android-studio-for-platform/common.nix:
```diff
@@ -0,0 +1,195 @@
+    dontPatchShebangs = true;
+
+    installPhase = ''
+      dpkg -x $src .
```
dpkg has a hook to do this

Artturin commented on pkgs/applications/editors/android-studio-for-platform/common.nix:
```diff
@@ -0,0 +1,195 @@
+
+    nativeBuildInputs = [
+      dpkg
+      unzip
```
Unzip here unnecessary?

robbins commented:
Also removed some un-needed dependencies in the latest push.
